core/domain/subtopic_validators_test.py

```diff
@@ -0,0 +1,857 @@
+
+        topics = [topic_domain.Topic.create_default_topic(
+            '%s' % i,
+            'topic%s' % i,
+            'abbrev-%s' % chr(120 + i),
+            'description%s' % i) for i in python_utils.RANGE(3)]
+        language_codes = ['ar', 'en', 'en']
+
+        for index, topic in enumerate(topics):
+            topic.language_code = language_codes[index]
+            topic.add_additional_story('%s' % (index * 2))
+            topic.add_canonical_story('%s' % (index * 2 + 1))
+            topic.add_uncategorized_skill_id('%s' % (index * 3))
+            topic.add_uncategorized_skill_id('%s' % (index * 3 + 1))
+            topic.add_uncategorized_skill_id('%s' % (index * 3 + 2))
+            topic_services.save_new_topic(self.owner_id, topic)
+            if index == 0:
+                committer_id = self.user_id
+            else:
+                committer_id = self.owner_id
+            topic_services.update_topic_and_subtopic_pages(
+                committer_id, '%s' % index, [topic_domain.TopicChange({
+                    'cmd': 'add_subtopic',
+                    'title': 'subtopic1',
+                    'subtopic_id': 1
+                }), topic_domain.TopicChange({
+                    'cmd': 'move_skill_id_to_subtopic',
+                    'old_subtopic_id': None,
+                    'new_subtopic_id': 1,
+                    'skill_id': '%s' % (index * 3)
+                }), topic_domain.TopicChange({
+                    'cmd': 'move_skill_id_to_subtopic',
+                    'old_subtopic_id': None,
+                    'new_subtopic_id': 1,
+                    'skill_id': '%s' % (index * 3 + 1)
+                })], 'Changes.')
+
+        self.model_instance_0 = (
+            subtopic_models.SubtopicPageSnapshotMetadataModel.get_by_id(
+                '0-1-1'))
+        self.model_instance_1 = (
+            subtopic_models.SubtopicPageSnapshotMetadataModel.get_by_id(
+                '1-1-1'))
+        self.model_instance_2 = (
+            subtopic_models.SubtopicPageSnapshotMetadataModel.get_by_id(
+                '2-1-1'))
+
+        self.job_class = (
+            prod_validation_jobs_one_off
+            .SubtopicPageSnapshotMetadataModelAuditOneOffJob)
+
+    def test_standard_operation(self):
+        topic_services.update_topic_and_subtopic_pages(
+            self.owner_id, '0', [subtopic_page_domain.SubtopicPageChange({
+                'cmd': 'update_subtopic_page_property',
+                'property_name': 'page_contents_html',
+                'subtopic_id': 1,
+                'new_value': {
+                    'html': '<p>html</p>',
+                    'content_id': 'content'
+                },
+                'old_value': {}
+            })], 'Changes.')
+        expected_output = [
+            u'[u\'fully-validated SubtopicPageSnapshotMetadataModel\', 4]']
+        self.run_job_and_check_output(
+            expected_output, sort=False, literal_eval=False)
```

Comment:
def test_model_with_committer_id_migration_bot(self):
        self.model_instance_1.committer_id = feconf.MIGRATION_BOT_USER_ID
        self.model_instance_1.update_timestamps(update_last_updated_time=False)
        self.model_instance_1.put()
        expected_output = [
            u'[u\'fully-validated SubtopicPageSnapshotMetadataModel\', 3]'
        ]
        self.run_job_and_check_output(
            expected_output, sort=False, literal_eval=False)
    def test_model_with_pseudo_committer_id(self):
        self.model_instance_1.committer_id = self.PSEUDONYMOUS_ID
        self.model_instance_1.update_timestamps(update_last_updated_time=False)
        self.model_instance_1.put()
        expected_output = [
            u'[u\'fully-validated SubtopicPageSnapshotMetadataModel\', 3]'
        ]
        self.run_job_and_check_output(
            expected_output, sort=False, literal_eval=False)

Reply:
@Rupeshkoushik Does this still need any clarification from the reviewers? If yes, then please let us know in a follow up comment what is your particular concern here. Thanks!

Reply:
No, it is resolved